File: python/Colleen/Colleen.py
def f(s):
	# commentaire hors du main
	print(s.format(10, '	', 34, s))

# commentaire dans le main
s = "def f(s):{0:c}{1}# commentaire hors du main{0:c}{1}print(s.format(10, '	', 34, s)){0:c}{0:c}# commentaire dans le main{0:c}s = {2:c}{3}{2:c}{0:c}f(s)"
f(s)

File: C/Colleen/Colleen.c
#include <stdio.h>

/*
	commentaire hors du main
*/

void f(char *s)
{
	printf(s, 10, s, 34);
}

int main()
{
	/*
		commentaire dans le main
	*/
	char *s = "#include <stdio.h>%1$c%1$c/*%1$c	commentaire hors du main%1$c*/%1$c%1$cvoid f(char *s)%1$c{%1$c	printf(s, 10, s, 34);%1$c}%1$c%1$cint main()%1$c{%1$c	/*%1$c		commentaire dans le main%1$c	*/%1$c	char *s = %3$c%2$s%3$c;%1$c	f(s);%1$c}%1$c";
	f(s);
}
